packages/pwa-kit-dev/bin/pwa-kit-dev.js

```diff
@@ -284,13 +277,63 @@ const main = () => {
     program
         .command('test')
         .description('test the project')
-        .action((_, {args}) => {
+        .action((_, opts) => {
+            const {args} = opts
             const jest = p.join(require.resolve('jest'), '..', '..', '..', '.bin', 'jest')
             execSync(
                 `${jest} --passWithNoTests --maxWorkers=2${args.length ? ' ' + args.join(' ') : ''}`
             )
         })
 
+    program
+        .command('logs')
+        .description(`tail environment logs`)
+        // TODO: add a --tail flag and make -p optional. get the default from package.json
+        .option('-p, --project <project_slug>', 'the project slug', (val) => {
+            if (!(typeof val === 'string') && val.length > 0) {
+                throw new program.InvalidArgumentError(`"${val}" cannot be empty`)
+            } else {
+                return val
+            }
+        })
+        .requiredOption('-e, --environment <environment_slug>', 'the environment slug', (val) => {
+            if (!(typeof val === 'string') && val.length > 0) {
+                throw new program.InvalidArgumentError(`"${val}" cannot be empty`)
+            } else {
+                return val
+            }
+        })
+        .action(async (_, opts) => {
+            const {project, environment, cloudApiBase, credentialsFile} = opts.optsWithGlobals()
+            let credentials
+            try {
+                credentials = fse.readJsonSync(credentialsFile)
+            } catch (e) {
+                scriptUtils.fail(`Error reading settings: ${e}`)
+            }
+
+            const token = await scriptUtils.createToken(project, environment, cloudApiBase, credentials.api_key)
+            const url = new URL(cloudApiBase.replace('cloud', 'logs'))
```

Comment:
Probs want to validate that URL actually looks "cloud-ish" if you're going to replace like that!

Reply:
I'll add validation to the option definition itself 😀